packages/webpack-cli/src/plugins/dotenv-webpack-plugin.ts

```diff
@@ -0,0 +1,200 @@
+    const vars: EnvVariables = this.#initializeVars();
+
+    const { env, blueprint } = this.#getEnvs();
+
+    Object.keys(blueprint).forEach((key) => {
+      const value = Object.prototype.hasOwnProperty.call(vars, key) ? vars[key] : env[key];
+
+      if (typeof value === "undefined" || value === null || (!allowEmptyValues && value === "")) {
+        compilation.errors.push(new Error(`Missing environment variable: ${key}`));
+      } else {
+        vars[key] = value;
+      }
+    });
+
+    return vars;
+  }
+
+  #initializeVars(): EnvVariables {
+    return this.#config.systemvars
+      ? Object.fromEntries(Object.entries(process.env).map(([key, value]) => [key, value ?? ""]))
+      : {};
+  }
+
+  #getEnvs(): { env: EnvVariables; blueprint: EnvVariables } {
+    const { paths } = this.#config;
+
+    const env: EnvVariables = {};
+
+    (paths || ["./.env"]).forEach((path) =>
+      Object.assign(env, dotenv.parse(this.#loadFile(path || "./.env"))),
+    );
+
+    const blueprint: EnvVariables = env;
+
+    return { env, blueprint };
+  }
+
+  #formatData({
+    variables = {},
+    target,
+  }: {
+    variables: EnvVariables;
+    target: string;
+  }): Record<string, string> {
+    const { expand, prefixes } = this.#config;
+
+    const preprocessedVariables: EnvVariables = Object.keys(variables).reduce(
+      (obj: EnvVariables, key: string) => {
+        let value = variables[key];
+        if (expand) {
+          if (value.startsWith("\\$")) {
+            value = value.substring(1);
+          } else if (value.includes("\\$")) {
+            value = value.replace(/\\\$/g, "$");
+          } else {
+            value = interpolate(value, variables);
+          }
+        }
+        obj[key] = JSON.stringify(value);
+        return obj;
+      },
+      {},
+    );
+
+    const formatted: Record<string, string> = {};
+    prefixes?.forEach((prefix) => {
+      Object.entries(preprocessedVariables).forEach(([key, value]) => {
+        formatted[`${prefix}${key}`] = value;
+      });
+    });
+
+    const shouldStubEnv =
+      prefixes?.includes("process.env.") && this.#shouldStub({ target, prefix: "process.env." });
+    if (shouldStubEnv) {
+      formatted["process.env"] = '"MISSING_ENV_VAR"';
+    }
+
+    return formatted;
+  }
+
+  #shouldStub({
+    target: targetInput,
+    prefix,
+  }: {
+    target: string | string[] | undefined;
+    prefix: string;
+  }): boolean {
+    const targets: string[] = Array.isArray(targetInput) ? targetInput : [targetInput || ""];
+
+    return targets.every(
+      (target) =>
+        prefix === "process.env." &&
+        this.#config.ignoreStub !== true &&
+        (this.#config.ignoreStub === false ||
+          (!target.includes("node") && !isMainThreadElectron(target))),
+    );
+  }
+
+  #loadFile(filePath: string): string {
+    try {
+      const content = this.#inputFileSystem.readFileSync(filePath, "utf8");
+      return content;
+    } catch (err) {
+      return "{}";
```

Comment:
Please don't do it, we need to log this - `this.#logger.log("Unable to upload ${filepath} file due:\n ${err.toString()}");`